[bug] Homebrew versions are invalid

**Describe the bug**
Using a homebrew copy of the version command reports a stack trace.

**To Reproduce**
Steps to reproduce the behavior:
1. Install `v0.4.0` via homebrew
2. Run `pygmy-go` version
3. See error

**Expected behavior**
Version correctly identified and reported

**Output**
```
$ brew info pygmy-go
fubarhouse/pygmy-go/pygmy-go: stable 0.4.0
amazee.io's local development helper tool
https://github.com/fubarhouse/pygmy-go
/home/linuxbrew/.linuxbrew/Cellar/pygmy-go/0.4.0 (5 files, 11.4MB) *
  Built from source on 2020-06-12 at 18:39:37
From: https://github.com/fubarhouse/homebrew-pygmy-go/blob/master/Formula/pygmy-go.rb
$ pygmy-go version
panic: runtime error: index out of range [2] with length 1
goroutine 1 [running]:
github.com/fubarhouse/pygmy-go/service/library.Version(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/runner/work/pygmy-go/pygmy-go/service/library/version.go:65 +0xa6b
github.com/fubarhouse/pygmy-go/cmd.glob..func10(0xf5ede0, 0xf92ef8, 0x0, 0x0)
	/home/runner/work/pygmy-go/pygmy-go/cmd/version.go:36 +0x4c
github.com/spf13/cobra.(*Command).execute(0xf5ede0, 0xf92ef8, 0x0, 0x0, 0xf5ede0, 0xf92ef8)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xf5e360, 0x4420ea, 0xf22b60, 0xc000054778)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/fubarhouse/pygmy-go/cmd.Execute()
	/home/runner/work/pygmy-go/pygmy-go/cmd/root.go:55 +0x31
main.main()
	/home/runner/work/pygmy-go/pygmy-go/main.go:26 +0x20
```

**Additional context**
The following success criteria are part of the resolution of this bug:
 - GitHub Actions need to be actionable and verifiable on a local system.
 - Release documentation would be nice to simulate the GitHub Action off-platform.
 - Version tag tests need to be created and accounted for (visual in-test verification)

--- a/.github/workflows/goreleaser.yml
+++ b/.github/workflows/goreleaser.yml
@@ -28,6 +28,9 @@ jobs:
       -
         run: cat service/library/version.go | grep "COMMITTAG =" | grep -v Getenv
       -
+        name: Fetch Dependencies
+        run: go mod download
+      -
         name: Run GoReleaser
         uses: goreleaser/goreleaser-action@v2
         with:

--- a/.github/workflows/goreleaser.yml
+++ b/.github/workflows/goreleaser.yml
@@ -10,14 +10,13 @@ on:
 
 jobs:
   goreleaser:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-18.04
     steps:
       -
         name: Checkout
         uses: actions/checkout@v2
       -
-        name: Unshallow
-        run: git fetch --prune --unshallow
+        run: apt update && apt install git -y
       -
         name: Set up Go
         uses: actions/setup-go@v2
@@ -26,6 +25,8 @@ jobs:
       -
         name: Establish version for compiled files
         run: sed -i -e "s|COMMITTAG = \"\"|COMMITTAG = \"$GITHUB_REF\"|g" service/library/version.go
+      -
+        run: cat service/library/version.go | grep "COMMITTAG =" | grep -v Getenv
       -
         name: Run GoReleaser
         uses: goreleaser/goreleaser-action@v2

--- a/.github/workflows/goreleaser.yml
+++ b/.github/workflows/goreleaser.yml
@@ -23,10 +23,17 @@ jobs:
         with:
           go-version: 1.14
       -
+        name: Establish commit ref for compiled files
+        run: sed -i -e "s|COMMITSHA = \"\"|COMMITSHA = \"$GITHUB_SHA\"|g" service/library/version.go
+      -
         name: Establish version for compiled files
         run: sed -i -e "s|COMMITTAG = \"\"|COMMITTAG = \"$GITHUB_REF\"|g" service/library/version.go
       -
-        run: cat service/library/version.go | grep "COMMITTAG =" | grep -v Getenv
+        name: Echo the version reference
+        run: cat service/library/version.go | grep "COMMITTAG =" | grep "refs" | head -n1 | xargs
+      -
+        name: Echo the commit reference
+        run: cat service/library/version.go | grep "COMMITSHA =" | head -n1 | xargs
       -
         name: Fetch Dependencies
         run: go mod download

--- a/.github/workflows/goreleaser.yml
+++ b/.github/workflows/goreleaser.yml
@@ -16,7 +16,7 @@ jobs:
         name: Checkout
         uses: actions/checkout@v2
       -
-        run: apt update && apt install git -y
+        run: apt update && apt install git golang -y
       -
         name: Set up Go
         uses: actions/setup-go@v2

--- a/.github/workflows/goreleaser.yml
+++ b/.github/workflows/goreleaser.yml
@@ -16,7 +16,7 @@ jobs:
         name: Checkout
         uses: actions/checkout@v2
       -
-        run: apt update && apt install git golang -y
+        run: sudo apt update && sudo apt install git golang -y
       -
         name: Set up Go
         uses: actions/setup-go@v2

--- a/.goreleaser.yml
+++ b/.goreleaser.yml
@@ -2,10 +2,6 @@ env:
   - GO111MODULE=on
   - GOPROXY=https://gocenter.io
 
-before:
-  hooks:
-    - go mod download
-
 builds:
   - env:
       - CGO_ENABLED=0

--- a/.travis.yml
+++ b/.travis.yml
@@ -89,6 +89,7 @@ matrix:
       os: linux
       services: docker
       before_install:
+        - apt-get install go -y
         - curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
 
       script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -85,21 +85,6 @@ matrix:
         - ./pygmy-go-linux-x86 --config examples/pygmy.travis.yml down;
         - ./pygmy-go-linux-x86 --config examples/pygmy.travis.yml clean;
 
-    - name: GitHub Actions simulation
-      os: linux
-      services: docker
-      before_install:
-        - curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
-
-      script:
-        - act -l
-        - act push --verbose -P ubuntu-latest=nektos/act-environments-ubuntu:18.04 || true
-        - stat /github/workspace/dist/pygmy-go_darwin_amd64/pygmy-go
-        - stat /github/workspace/dist/pygmy-go_linux_amd64/pygmy-go
-        - stat /github/workspace/dist/pygmy-go_windows_amd64/pygmy-go.exe
-        - /github/workspace/dist/pygmy-go_linux_amd64/pygmy-go version
-        - /github/workspace/dist/pygmy-go_linux_amd64/pygmy-go status
-
     - name: Windows build
       os: windows
       services: docker

--- a/.travis.yml
+++ b/.travis.yml
@@ -89,7 +89,7 @@ matrix:
       os: linux
       services: docker
       before_install:
-        - sudo apt-get install go -y
+        - sudo apt-get install golang-go -y
         - curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
 
       script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -89,6 +89,7 @@ matrix:
       os: linux
       services: docker
       before_install:
+        - go mod vendor
         - curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
 
       script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -89,17 +89,16 @@ matrix:
       os: linux
       services: docker
       before_install:
-        - go mod vendor
         - curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
 
       script:
         - act -l
-        - act push --verbose || true
+        - act push --verbose -P ubuntu-latest=nektos/act-environments-ubuntu:18.04 || true
         - stat /github/workspace/dist/pygmy-go_darwin_amd64/pygmy-go
         - stat /github/workspace/dist/pygmy-go_linux_amd64/pygmy-go
         - stat /github/workspace/dist/pygmy-go_windows_amd64/pygmy-go.exe
-        - /github/workspace//dist/pygmy-go_linux_amd64/pygmy-go version
-        - /github/workspace//dist/pygmy-go_linux_amd64/pygmy-go status
+        - /github/workspace/dist/pygmy-go_linux_amd64/pygmy-go version
+        - /github/workspace/dist/pygmy-go_linux_amd64/pygmy-go status
 
     - name: Windows build
       os: windows

--- a/.travis.yml
+++ b/.travis.yml
@@ -85,6 +85,20 @@ matrix:
         - ./pygmy-go-linux-x86 --config examples/pygmy.travis.yml down;
         - ./pygmy-go-linux-x86 --config examples/pygmy.travis.yml clean;
 
+    - name: GitHub Actions simulation
+      os: linux
+      services: docker
+      before_install:
+        - curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
+
+      script:
+        - act -l
+        - act push --verbose || true
+        - stat ./dist/pygmy-go_darwin_amd64/pygmy-go
+        - stat ./dist/pygmy-go_linux_amd64/pygmy-go
+        - stat ./dist/pygmy-go_windows_amd64/pygmy-go.exe
+        - ./dist/pygmy-go_linux_amd64/pygmy-go version
+        - ./dist/pygmy-go_linux_amd64/pygmy-go status
 
     - name: Windows build
       os: windows

--- a/.travis.yml
+++ b/.travis.yml
@@ -89,7 +89,7 @@ matrix:
       os: linux
       services: docker
       before_install:
-        - apt-get install go -y
+        - sudo apt-get install go -y
         - curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
 
       script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -89,17 +89,16 @@ matrix:
       os: linux
       services: docker
       before_install:
-        - sudo apt-get install golang-go -y
         - curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
 
       script:
         - act -l
         - act push --verbose || true
-        - stat ./dist/pygmy-go_darwin_amd64/pygmy-go
-        - stat ./dist/pygmy-go_linux_amd64/pygmy-go
-        - stat ./dist/pygmy-go_windows_amd64/pygmy-go.exe
-        - ./dist/pygmy-go_linux_amd64/pygmy-go version
-        - ./dist/pygmy-go_linux_amd64/pygmy-go status
+        - stat /github/workspace/dist/pygmy-go_darwin_amd64/pygmy-go
+        - stat /github/workspace/dist/pygmy-go_linux_amd64/pygmy-go
+        - stat /github/workspace/dist/pygmy-go_windows_amd64/pygmy-go.exe
+        - /github/workspace//dist/pygmy-go_linux_amd64/pygmy-go version
+        - /github/workspace//dist/pygmy-go_linux_amd64/pygmy-go status
 
     - name: Windows build
       os: windows

--- a/README.md
+++ b/README.md
@@ -81,6 +81,6 @@ Thanks for testing, please post issues and successes in the queue.
  
 ## Releasing
  
-We use GitHub Actions for automated release tagging. Using [Act](https://github.com/nektos/act) locally, you can simulate this process and have the same build artifacts in your `dist` folder.
+We use GitHub Actions for simulating the automated release tagging locally. Using [Act](https://github.com/nektos/act) locally, you can simulate this process and have the same build artifacts in your `dist` folder.
  
 This process will inject the appropriate values into the version logic. To start the process, just run `act`!

--- a/README.md
+++ b/README.md
@@ -79,3 +79,8 @@ Content-Type: text/html
 
 Thanks for testing, please post issues and successes in the queue.
  
+## Releasing
+ 
+We use GitHub Actions for automated release tagging. Using [Act](https://github.com/nektos/act) locally, you can simulate this process and have the same build artifacts in your `dist` folder.
+ 
+This process will inject the appropriate values into the version logic. To start the process, just run `act`!

--- a/service/library/version.go
+++ b/service/library/version.go
@@ -12,7 +12,7 @@ var (
 	// Version tag used on local builds in Actions.
 	COMMITSHA = os.Getenv("GITHUB_SHA")
 
-	// Fixed version which is modified via travis for the
+	// Fixed version which is modified via github actions for the
 	// release builds. Should match version with v prepended.
 	COMMITTAG = ""
 
@@ -28,8 +28,7 @@ var (
 func Version(c Config) {
 
 
-	// RELEASETAG and VERSIONTAG should be set when running Actions.
-	// Travis will add a commit has which should not print in this section.
+	// VERSIONTAG should be set when running Actions.
 	if COMMITTAG != "" {
 		COMMITTAG = strings.TrimLeft(COMMITTAG, "refs/tags/")
 		reference := ""

--- a/service/library/version.go
+++ b/service/library/version.go
@@ -38,7 +38,7 @@ func Version(c Config) {
 	if COMMITSHA == "" {
 		r, _ := exec.Command("git", "remote", "-v").Output()
 		remotes := strings.Split(string(r), "\n")
-		for remote, _ := range remotes {
+		for remote := range remotes {
 			if strings.Contains(remotes[remote], "pygmy-go") {
 				isPygmy = true
 			}

--- a/service/library/version.go
+++ b/service/library/version.go
@@ -55,7 +55,6 @@ func Version(c Config) {
 
 	// Detect a tagged version:
 	if !isPygmy && COMMITTAG != "" {
-		COMMITTAG = strings.TrimLeft(COMMITTAG, "refs/tags/")
 		b := strings.Split(COMMITTAG, "/")
 		reference := b[len(b)-1]
 

--- a/service/library/version.go
+++ b/service/library/version.go
@@ -27,9 +27,6 @@ var (
 // after the release is published.
 func Version(c Config) {
 
-	if COMMITSHA == "" {
-		COMMITTAG = os.Getenv("TRAVIS_COMMIT")
-	}
 
 	// RELEASETAG and VERSIONTAG should be set when running Actions.
 	// Travis will add a commit has which should not print in this section.
@@ -47,6 +44,11 @@ func Version(c Config) {
 			fmt.Printf("Pygmy version dev-%v\n", reference[0:7])
 			return
 		}
+	}
+
+	if COMMITSHA == "" {
+		// Fallback to TRAVIS_COMMIT if GITHUB_SHA is absent.
+		COMMITTAG = os.Getenv("TRAVIS_COMMIT")
 	}
 
 	// Get tags and reference information.

--- a/service/library/version.go
+++ b/service/library/version.go
@@ -31,6 +31,7 @@ func Version(c Config) {
 	// RELEASETAG and VERSIONTAG should be set when running Actions.
 	// Travis will add a commit has which should not print in this section.
 	if COMMITTAG != "" {
+		COMMITTAG = strings.TrimLeft(COMMITTAG, "refs/tags/")
 		reference := ""
 		if strings.Contains(COMMITTAG, "/") {
 			reference = strings.Split(COMMITTAG, "/")[2]

--- a/service/library/version.go
+++ b/service/library/version.go
@@ -2,15 +2,20 @@ package library
 
 import (
 	"fmt"
-	"os"
 	"os/exec"
 	"regexp"
 	"strings"
 )
 
 var (
+
+	// isPygmy will determine if the local git branch is Pygmy.
+	// otherwise the mechanisms to allow non-pygmy hash refs
+	// being shown.
+	isPygmy = false
+
 	// Version tag used on local builds in Actions.
-	COMMITSHA = os.Getenv("GITHUB_SHA")
+	COMMITSHA = ""
 
 	// Fixed version which is modified via github actions for the
 	// release builds. Should match version with v prepended.
@@ -27,53 +32,46 @@ var (
 // after the release is published.
 func Version(c Config) {
 
-
-	// VERSIONTAG should be set when running Actions.
-	if COMMITTAG != "" {
-		COMMITTAG = strings.TrimLeft(COMMITTAG, "refs/tags/")
-		reference := ""
-		if strings.Contains(COMMITTAG, "/") {
-			reference = strings.Split(COMMITTAG, "/")[2]
-		} else {
-			reference = COMMITTAG
-		}
-		if match, _ := regexp.Match("^v[0-9]+.[0-9]+.[0-9]+$", []byte(reference)); match {
-			fmt.Printf("Pygmy %v\n", reference)
-			return
-		} else if match, _ := regexp.Match("^[0-9|a-z|A-Z]+$", []byte(reference)); match {
-			fmt.Printf("Pygmy version dev-%v\n", reference[0:7])
-			return
-		}
-	}
-
+	// The Commit SHA is provided to GitHub Actions, if it is not set when compiled
+	// we assume git is being used. Let's grab metadata from the repository for use
+	// in the runtime -
 	if COMMITSHA == "" {
-		// Fallback to TRAVIS_COMMIT if GITHUB_SHA is absent.
-		COMMITTAG = os.Getenv("TRAVIS_COMMIT")
+		r, _ := exec.Command("git", "remote", "-v").Output()
+		remotes := strings.Split(string(r), "\n")
+		for remote, _ := range remotes {
+			if strings.Contains(remotes[remote], "pygmy-go") {
+				isPygmy = true
+			}
+		}
+
+		sha, _ := exec.Command("git", "rev-parse", "HEAD").Output()
+		COMMITSHA = string(sha)
+
+		_, changes := exec.Command("git", "diff-index", "--quiet", "HEAD").Output()
+		if changes != nil {
+			CUSTOMTAG = "-custom"
+		}
 	}
 
-	// Get tags and reference information.
-	tags, _ := exec.Command("git", "show-ref", "--tags").Output()
-	ref, _ := exec.Command("git", "rev-parse", "HEAD").Output()
-	_, changes := exec.Command("git", "diff-index", "--quiet", "HEAD").Output()
-	if changes != nil {
-		CUSTOMTAG = "-custom"
-	}
+	// Detect a tagged version:
+	if !isPygmy && COMMITTAG != "" {
+		COMMITTAG = strings.TrimLeft(COMMITTAG, "refs/tags/")
+		b := strings.Split(COMMITTAG, "/")
+		reference := b[len(b)-1]
 
-	// Scan the references from the tags to check if the current reference is
-	// associated to a tag. This is to show the tag version on a local build
-	// from source.
-	for _, tag := range strings.Split(string(tags), "\n") {
-		if strings.Contains(tag, string(ref)) {
-			fmt.Printf("Pygmy %v%v\n", strings.SplitAfter(tag, "/")[2], CUSTOMTAG)
+		if match, _ := regexp.Match("^v[0-9]+.[0-9]+.[0-9]+$", []byte(reference)); match {
+			fmt.Printf("Pygmy %v%v\n", reference, CUSTOMTAG)
 			return
 		}
 	}
 
-	if COMMITTAG != "" {
-		// If the version tag isn't empty:
-		fmt.Printf("Pygmy %v%v\n", COMMITTAG, CUSTOMTAG)
+	// Detect a SHA reference:
+	if isPygmy || COMMITSHA != "" {
+		if match, _ := regexp.Match("^[0-9|a-z|A-Z]+$", []byte(COMMITSHA)); match {
+			fmt.Printf("Pygmy version dev-%v%v\n", COMMITSHA[0:7], CUSTOMTAG)
+			return
+		}
 	} else {
-		// If we don't have a version tag, use a reference.
-		fmt.Printf("Pygmy version dev-%v%v\n", string(ref[0:7]), CUSTOMTAG)
+		fmt.Printf("Pygmy version unidentifiable.\n")
 	}
 }

